src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -48,8 +48,9 @@ internal class SolutionBindingOperation : ISolutionRuleStore
         private Dictionary<Language, SonarQubeQualityProfile> qualityProfileMap;
         private readonly ConnectionInformation connection;
         private readonly string ProjectKey;
+        private readonly SonarLintMode bindingMode;
 
-        public SolutionBindingOperation(IServiceProvider serviceProvider, ConnectionInformation connection, string ProjectKey)
+        public SolutionBindingOperation(IServiceProvider serviceProvider, ConnectionInformation connection, string ProjectKey, SonarLintMode bindingMode)
```

Comment:
I took the simple approach of making this class aware of the binding mode and put in conditional checks in a couple of places - I couldn't see an easy way to factor out the mode-specific code in the way I did for the _BindingWorkflow_.